pilot/pkg/serviceregistry/kube/controller/ambientindex_external.go

```diff
@@ -562,13 +562,14 @@ func (a *AmbientIndexImpl) cleanupOldWorkloadEntriesInlinedOnServiceEntry(svcEnt
 	if oldServiceEntry, f := a.servicesMap[nsName]; f {
 		for _, oldWe := range oldServiceEntry.Spec.Endpoints {
 			oldUID := c.generateServiceEntryUID(nsName.Namespace, nsName.Name, oldWe.Address)
-			we, found := a.byUID[oldUID]
-			if found {
+			if we, found := a.byUID[oldUID]; found {
 				updates.Insert(model.ConfigKey{Kind: kind.Address, Name: we.ResourceName()})
 				for _, networkAddr := range networkAddressFromWorkload(we) {
 					delete(a.byWorkloadEntry, networkAddr)
 				}
 				delete(a.byUID, oldUID)
+				weUID := c.generateWorkloadEntryUID(nsName.Namespace, nsName.Name)
```

Comment:
why do we generate WLE UID from SE, and where do we construct and store it

Reply:
In L122, we generate with address

```
			a.byUID[c.generateServiceEntryUID(svcEntry.GetNamespace(), svcEntry.GetName(), we.GetAddress())] = wli

```

Reply:
The call in line https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/kube/controller/ambientindex_external.go#L94 returns Workload Entries that include the inlined

Reply:
Confused, `c.getSelectedWorkloadEntries(svcEntry.GetNamespace(), svcEntry.Spec.GetWorkloadSelector().GetLabels())` only returns the separate WLE objects.

Here we cache the inlined wle endpoints https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/kube/controller/ambientindex_external.go#L116-L125

Reply:
The function also returns inlined. See: https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/kube/controller/ambientindex_external.go#L506

Reply:
Maybe i missed something before, i donot think `c.getSelectedWorkloadEntries(svc.GetNamespace(), svc.Spec.Selector)` should select a ServiceEntry's endpoint

Reply:
@hzxuzhonghu I believe I targeted your concern about the `getSelectedWorkloadEntries`. It needs to be selected only for policies.

The inlined WLE is added to the index here: https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/kube/controller/authorization.go#L497
and we need to clean it up upon deletion.